### PR TITLE
HV-1971 fix CNPJ validation when all digits are the same

### PR DESCRIPTION
**Pull request by edurbs**
https://hibernate.atlassian.net/browse/HV-1971

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/br/CNPJValidator.java [hunk 1]
```diff
@@ -55,6 +56,13 @@ public boolean isValid(CharSequence value, ConstraintValidatorContext context) {
 			return true;
 		}
 
+		// Check for repeated digits
+		if ( value.toString().chars()
```
Comment (marko-bekhta) [inline engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/br/CNPJValidator.java:60]: Maybe let's replace this with a simple for loop? Something along the lines:
```java
// the number is too short, hence it's invalid anyway:
if ( value.length() < 2 ) {
	return false;
}

char firstDigit = value.charAt( 0 );
char otherDigit = value.charAt( 1 );
for ( int i = 2; i < value.length(); i++ ) {
	char c = value.charAt( i );
	if ( Character.isDigit( c ) && firstDigit != c ) {
		otherDigit = c;
	}
}
if ( firstDigit == otherDigit ) {
	return false;
}
```

From what I see `distinct` may not be optimized:
```java
public final IntStream distinct() {
	// While functional and quick to implement, this approach is not very efficient.
	// An efficient version requires an int-specific map/set implementation.
	return boxed().distinct().mapToInt(i -> i);
}
```

**Review by marko-bekhta — COMMENTED** [review]
Hey, thanks for reporting the problem and submitting a PR. Please see an inline comment.

File: engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/br/CNPJValidator.java (hunk 1 above)
Comment (edurbs) [reply engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/br/CNPJValidator.java:60]: perfect, I'll change it. Thanks!

**Review by marko-bekhta — APPROVED** [review]
Looks good, thanks!

**Pull request merged**